.github/workflows/docker-image-mariadb103-py38-mysqlclient201.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Docker Image CI mariadb103-py38-mysqlclient201
+
+on:
+  push:
+    paths:
+      - 'test-containers/mariadb103-py38-mysqlclient201/**'
+      - '.github/workflows/docker-image-mariadb103-py38-mysqlclient201.yml'
+      - '.github/workflows/build-docker-image.yml'
+
+jobs:
+
+  call-workflow-passing-data:
```

Comment:
@laurent-indermuehle why did you go for multiple reusable workflow invocations? This causes a lot of duplication, which is prone to human error.

Besides, multiple separate workflows may fight for the resources more, aren't consolidated visually and make it hard to have unified merge gates…

Have you tried wrapping this into a matrix? I'm pretty sure this would work combined with reusable workflows.

Reply:
@webknjaz you raise a good question. I think my main goal was to rebuild the containers only if needed. Do you think it would be possible to combine a matrix and an "on: push: paths + branch_ignore: stable-*" ?

Reply:
Yeah, it's 💯 possible with an extra small decision-making job that calculates conditions or even better — generates the matrix. I've been meaning to make an example PR but it fell off my radar. Maybe, later...

Reply:
@webknjaz I'm finally attempting to simplify the build of the test containers as you suggested. I've done a first attempt in #643. I would love to have your input on this if you have the time.